File: src/2482.cpp
#include <vector>

using namespace std;

class Solution {
public:
    vector<vector<int>> onesMinusZeros(vector<vector<int>>& grid) {
        int m = grid[0].size();
        int n = grid.size();
        vector<vector<int>> ret(n, vector<int>(m, 0));
        vector<int> rows (n, 0);
        vector<int> cols (m, 0);
        for (int r = 0; r < n; ++r) {
            for (int c = 0; c < m; ++c) {
                if (grid[r][c] != 0) {
                    rows[r]++;
                }
            }
        }
        for (int c = 0; c < m; ++c) {
            for (int r = 0; r < n; ++r) {
                if (grid[r][c] != 0) {
                    cols[c]++;
                }
            }
        }
        for (int r = 0; r < n; ++r) {
            for (int c = 0; c < m; ++c) {
                ret[r][c] = 2 * (rows[r] + cols[c]) - (n + m);
            }
        }
        return ret;
    }
};

File: src/2482.py
class Solution:
    def onesMinusZeros(self, grid: list[list[int]]) -> list[list[int]]:
        m = len(grid[0])
        n = len(grid)
        ret = [[0 for _ in range(m)] for _ in range(n)]
        rows = list(map(sum, grid))
        cols = [0 for _ in range(m)]
        for c in range(m):
            for r in range(n):
                if grid[r][c] != 0:
                    cols[c] += 1
        for r in range(n):
            for c in range(m):
                ret[r][c] = 2 * (rows[r] + cols[c]) - (n + m)
        return ret
